src/ORM/DataExtension.php

```diff
@@ -60,42 +63,116 @@ public function augmentWrite(&$manipulation)
     {
     }
 
+    /**
+     * Extend the owner's onBeforeWrite() logic
+     *
+     * See {@link DataObject::onBeforeWrite()} for context.
+     */
     public function onBeforeWrite()
     {
     }
 
+    /**
+     * Extend the owner's onAfterWrite() logic
+     *
+     * See {@link DataObject::onAfterWrite()} for context.
+     */
     public function onAfterWrite()
     {
     }
 
+    /**
+     * Extend the owner's onBeforeDelete() logic
+     *
+     * See {@link DataObject::onBeforeDelete()} for context.
+     */
     public function onBeforeDelete()
     {
     }
 
+    /**
+     * Extend the owner's onAfterDelete() logic
+     *
+     * See {@link DataObject::onAfterDelete()} for context.
+     */
     public function onAfterDelete()
     {
     }
 
+    /**
+     * Extend the owner's requireDefaultRecords() logic
+     *
+     * See {@link DataObject::requireDefaultRecords()} for context.
+     */
     public function requireDefaultRecords()
     {
     }
 
+    /**
+     * Extend the owner's populateDefaults() logic
+     *
+     * See {@link DataObject::populateDefaults()} for context.
+     */
     public function populateDefaults()
     {
     }
 
+    /**
+     * Influence the owner's can() permission check value to be disallowed (false),
+     * allowed (true) if no other processed results are to disallow, or open (null) to not
+     * affect the outcome.
+     *
+     * See {@link DataObject::can()} and {@link DataObject::extendedCan()} for context.
+     *
+     * @param Member $member
+     * @param array $context
+     * @return bool
```

Comment:
```suggestion
     * @return bool|null
```

Those of us with detailed IDEs might get annoyed at the indicated errors from this. This suggestion can be applied to all the `can*` methods.

Reply:
Oh? I assumed this was an anti-pattern because I see it so rarely in the core DocBlocks. Is there an internal convention to type-hint all outcomes?

Reply:
I'm not aware that this is an anti-pattern. I mean, having nullable return types in general might be considered one, but documenting it given it already exists surely isn't. I would be pretty against that notion too 😅

Reply:
We used to not declare null as an option, but we need to start being stricter as we move towards strict typing, so I’m with ScopeyNZ here

Reply:
Awesome. Great suggestion, thanks @ScopeyNZ 😃